### PR TITLE
 Homework for hyf_database week_1 added

### PR DESCRIPTION
**Pull request by basafilm**

### CODE REVIEW

File: databases/week1/hyf_database_week_1.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+WHERE title LIKE '%database%' OR description like '%database%';
+
+-- Get the title and status (as text) of all tasks
+SELECT  CONCAT(title, status_id) FROM task; 
+ 
+-- Get the name of each status, along with a count of how many tasks have that status
+SELECT status.id, status.name, COUNT(task.id) AS task_count FROM status
+LEFT JOIN task ON status.id = task.user_id
+GROUP BY status.id
+HAVING task_count >1 ;
```
Comment (flemeur) [inline databases/week1/hyf_database_week_1.sql:27]: You are very close to the correct solution here.

Two things:
Make sure you are choosing the correct foreign key in the ON clause of the JOIN - Can you spot the mistake?
Secondly you should not need the `HAVING task_count >1`. We would actually like to also include the statuses even if the count is zero.

File: databases/week1/hyf_database_week_1.sql [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+-- Get the names of all statuses, sorted by the status with most tasks first
+SELECT  status.name, task.id AS task_name  
+FROM status
+LEFT JOIN task ON status.id = task.user_id WHERE task.status_id > 1
+ORDER BY task_name DESC;
```
Comment (flemeur) [inline databases/week1/hyf_database_week_1.sql:33]: This one builds on the previous query so they should be almost identical. Try fixing the previous query first and then this one should be a lot easier to get correct :)

File: databases/week1/hyf_database_week_1.sql [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+-- Get the title and status (as text) of all tasks
+SELECT task.title, status.name FROM task 
+INNER JOIN status ON task.status_id = status.id;
+
+-- Get the name of each status, along with a count of how many tasks have that status
+SELECT status.id, status.name, COUNT(task.id) AS task_count 
+FROM task
+LEFT JOIN status ON status.id = status.id
+GROUP BY status.id;
```
Comment (flemeur) [inline databases/week1/hyf_database_week_1.sql:36]: You are so close with this one. I think you might just have introduced a typo by mistake. Can you spot the mistake?

File: databases/week1/hyf_database_week_1.sql [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+-- Get the names of all statuses, sorted by the status with most tasks first
+SELECT  status.name, task.id AS task_name  
+FROM status
+LEFT JOIN task ON status.id = task.user_id 
+ORDER BY task_name DESC;
```
Comment (flemeur) [inline databases/week1/hyf_database_week_1.sql:42]: This one should look almost exactly like the previous, except it also needs to order by the task count.

Comment (basafilm) [reply databases/week1/hyf_database_week_1.sql]: Thank You I think I got. it.

**Pull request merged**